File: utils/storage_config.py
"""Contact storage backend selection via CONTACT_STORAGE env."""
import os


VALID_STORAGE_MODES = ("postgresql", "firebase", "indexeddb")


def get_contact_storage_mode() -> str:
    raw = os.getenv("CONTACT_STORAGE", "postgresql").strip().lower()
    if raw in VALID_STORAGE_MODES:
        return raw
    return "postgresql"


def is_client_side_storage() -> bool:
    return get_contact_storage_mode() == "indexeddb"

File: src/lib/storageConfig.ts
export type ContactStorageMode = "postgresql" | "firebase" | "indexeddb";

const VALID: ContactStorageMode[] = ["postgresql", "firebase", "indexeddb"];

/** Must match backend CONTACT_STORAGE. Default: postgresql */
export function getContactStorageMode(): ContactStorageMode {
  const raw = String(import.meta.env.VITE_CONTACT_STORAGE || "postgresql")
    .trim()
    .toLowerCase();
  if (VALID.includes(raw as ContactStorageMode)) {
    return raw as ContactStorageMode;
  }
  return "postgresql";
}

export function isIndexedDbStorage(): boolean {
  return getContactStorageMode() === "indexeddb";
}

export function isServerStorage(): boolean {
  return getContactStorageMode() !== "indexeddb";
}

export function storageLabel(mode = getContactStorageMode()): string {
  switch (mode) {
    case "indexeddb":
      return "IndexedDB";
    case "firebase":
      return "Firebase";
    default:
      return "PostgreSQL";
  }
}
